src/abstractdataframe/join.jl

```diff
@@ -212,8 +212,34 @@ function update_row_maps!(left_table::AbstractDataFrame,
     return to_bimap(left_ixs), to_bimap(leftonly_ixs), to_bimap(right_ixs), to_bimap(rightonly_ixs)
 end
 
+function check_uniqueness_func(check_uniqueness, joiner)
+
+    if check_uniqueness == (true, false) && any(nonunique(joiner.dfl, joiner.left_on))
```

Comment:
I'd simplify this code by handling `df1` and `df2` separately (if `check_uniqueness` is false then just do as if the check was OK), and only then have an `if` with three branches determining the error message to print.

Since you compute `nonunique`, it would also be nice to give the name of the first row for which the problem was found. It's very useful when trying to find out the problem.

Reply:
I split it out to minimize computational costs. This way we don't ever run a `nonunique` unnecessarily. If I do `df1` and `df2` up front, we run it on both even if we only need to check one. 

> Since you compute nonunique, it would also be nice to give the name of the first row for which the problem was found. It's very useful when trying to find out the problem.

Good idea. Will put in error message.

Reply:
Yes but you can do something like:
```julia
left_valid = check_uniqueness[1] ? any(nonunique(joiner.dfl, joiner.left_on)) : true
right_valid = check_uniqueness[2] ? any(nonunique(joiner.dfr, joiner.right_on)) : true
```

Reply:
ok